subsys/net/lib/nrf_cloud/include/nrf_cloud_net_utils.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* Header file for internal nrf_cloud net lib utils. */
+
+#ifndef NRF_CLOUD_NET_UTIL_H_
+#define NRF_CLOUD_NET_UTIL_H_
+
+#include <zephyr/kernel.h>
+
+#if defined(CONFIG_POSIX_API)
+#include <zephyr/posix/sys/socket.h>
+#else
+#include <zephyr/net/socket.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @cond INTERNAL_HIDDEN */
+
+/** @brief Callback called for every address tried by nrf_cloud_connect_host.
+ * Should return either a valid socket, or a negative error code.
+ */
+typedef int (*nrf_cloud_try_address_cb)(struct sockaddr *const addr);
```

Comment:
```suggestion
typedef int (* nrf_cloud_connect_host_cb)(struct sockaddr *const addr);
```

i realize this is an internal header... but still, `nrf_cloud_try_address` is a static function.
`nrf_cloud_connect_host` is the "public" function, so i would prefer the callback name reference it.

Reply:
Valid point, I will switch to that